File: scala/src/main/scala/com/spoonacular/client/model/InlineObject5.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import java.math.BigDecimal

/**
 * 
 * @param username The username.
 * @param id The shopping list item id.
 * @param hash The private hash for the username.
 */
case class InlineObject5(username: String,
                id: BigDecimal,
                hash: String
                )

object InlineObject5 {
    /**
     * Creates the codec for converting InlineObject5 from and to JSON.
     */
    implicit val decoder: Decoder[InlineObject5] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineObject5] = deriveEncoder
}

File: dart/com/spoonacular/client/model/inline_object5.dart
part of openapi.api;

class InlineObject5 {
  /* The username. */
  String username = null;
  /* The shopping list item id. */
  num id = null;
  /* The private hash for the username. */
  String hash = null;
  InlineObject5();

  @override
  String toString() {
    return 'InlineObject5[username=$username, id=$id, hash=$hash, ]';
  }

  InlineObject5.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['username'] == null) {
      username = null;
    } else {
          username = json['username'];
    }
    if (json['id'] == null) {
      id = null;
    } else {
          id = json['id'];
    }
    if (json['hash'] == null) {
      hash = null;
    } else {
          hash = json['hash'];
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (username != null)
      json['username'] = username;
    if (id != null)
      json['id'] = id;
    if (hash != null)
      json['hash'] = hash;
    return json;
  }

  static List<InlineObject5> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineObject5>() : json.map((value) => new InlineObject5.fromJson(value)).toList();
  }

  static Map<String, InlineObject5> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineObject5>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineObject5.fromJson(value));
    }
    return map;
  }
}

